internal/commands/profile/init.go

```diff
@@ -62,16 +79,65 @@ type InitOpts struct {
 	IAMClient           iam_service.ClientService
 	OrganizationService organization_service.ClientService
 	ProjectService      project_service.ClientService
+	SecretService       preview_secret_service.ClientService
+
+	// Flags
+	VaultSecrets bool
 }
 
 func (i *InitOpts) run() error {
 	if err := i.configureOrgAndProject(); err != nil {
 		return fmt.Errorf("failed configuring organization and project: %w", err)
 	}
 
+	err := i.serviceConfigPrompt()
+	if err != nil {
+		return err
+	}
+
+	if i.VaultSecrets {
+		if err := i.configureVaultSecrets(); err != nil {
+			return fmt.Errorf("failed configuring profile for vault secrets: %w", err)
+		}
+	}
+
 	return i.Profile.Write()
 }
 
+func (i *InitOpts) serviceConfigPrompt() error {
+	if i.VaultSecrets {
+		return nil
+	}
+
+	if i.IO.CanPrompt() {
```

Comment:
This isn't needed since the command checks prompting is enabled: https://github.com/hashicorp/hcp/pull/64/files#diff-aae0cdd4820c4e0243986e34d1dd29c5014d2b240bef4b360738cf7e8ff1238cR53

Reply:
I think why I added is coz current PromptConfirm raises an error, but I was trying to bypass promoting if prompt is disabled.